src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexParser.cs

```diff
@@ -420,7 +423,7 @@ private RegexNode ScanRegex()
                             break;
 
                         default:
-                            throw MakeException(SR.InternalError);
+                            throw MakeException(RegexParseError.InternalError, SR.InternalError);
```

Comment:
this doesn't seem like this should be a parse/argument exception.  this should just be something like an InvalidOperationException.

Reply:
Technically you literally should not be able to hit this.  This is an internal error in your code.  It's the difference between improper input from teh user, versus a bug in your own code.

It's no more of a breaking change than the change you made to stop throwing an 'index out of bounds error' based off of the bugs i filed.  Technically someone could have been dependent on internal details about how your code crashes... but i don't think .net ever tries to preserve internal crashing behavior.

Reply:
I ran all the existing tests + the roslyn ones and never hit the internal error. I agree with @CyrusNajmabadi that changing to InvalidOperationException make more sense here. Are you ok with that Dan?

Reply:
Up to you, but I could turn this on its head and say that if we never throw it, it doesn't matter enough to change the type!

Reply:
I just read through the code and both places where the internal errors are thrown can never be reached unless someone changes the detection (the helper functions) of what a quantifier or a special character is. We shouldn't bother too much with that.